hack/build-local-images.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh -e
+# This script updates infrastructure images with a new "openshift" binary from your build workspace.
+# Only infrastructure images that run the openshift binary (directly or via symlink) are updated.
+# After running this, you can use the new images via "oc cluster up --version=latest" or
+# "openshift start --latest-images=true"
+# You can filter which images are built by passing one or more image name arguments, e.g.:
+# "hack/build-local-images.sh docker-builder sti-builder"
+# Only exact matching of image names (minus the ose/origin prefix) is supported.
+
+#imagenames=(docker-registry origin deployer recycler docker-builder                sti-builder                f5-router node openvswitch docker-registry cluster-capacity federation)
+#imagedirs=( dockerregistry  origin deployer recycler builder/docker/docker-builder builder/docker/sti-builder router/f5 node openvswitch docker-registry cluster-capacity federation)
+imagenames=(origin deployer recycler docker-builder                sti-builder                f5-router node openvswitch)
+imagedirs=( origin deployer recycler builder/docker/docker-builder builder/docker/sti-builder router/f5 node openvswitch)
+
+count=${#imagenames[@]}
+crc_count=${#imagedirs[@]}
+if [ $count -ne $crc_count ]; then
+    echo "imagenames and imagedirs arrays are different sizes, they need to be parallel arrays."
+    echo "$count $crc_count"
+    exit 1
+fi
+
+# reuse the same bin directory for all the docker builds
+# to save time copying stuff around.
+BIN_TMP=`mktemp -d`
+echo "Staging bin files in $BIN_TMP"
+#mkdir -p $BIN_TMP/bin
+#cp -a `pwd`/_output/local/bin/linux/amd64/* $BIN_TMP/bin
+ln -s `pwd`/_output/local/bin/linux/amd64 $BIN_TMP/bin
+
+pushd $BIN_TMP/bin
+
+trap cleanup exit
+function cleanup {
+    popd
+    rm -rf $BIN_TMP    
+}
+
+c=-1
+while [ $c -lt $((count-1)) ]; do 
+    c=$((c+1))
+    imagename=${imagenames[$c]}
+    imagedir=${imagedirs[$c]}
+    tag_prefix="openshift/origin-"
+    if [[ $imagename == "origin" || $imagename == "node" || $imagename ==  "openvswitch" ]]; then
+        tag_prefix="openshift/"
+    fi
+
+
+    # for most images we are replacing the openshift binary, but for the
+    # docker registry image, we're replacing the dockerregistry binary.
+    # similar hacks for other images.
+    filename="openshift"
+    #if [ $imagename == "docker-registry" ]; then
```

Comment:
i commented it out because it's currently broken for this script.

the script as it stands at the moment can only add files that exist in the output dir, which does not appear to contain the dockerregistry binary.  

Images which need content from locations other than the standard output dir are going to require special logic in this script to setup a docker build context dir properly.